### PR TITLE
Added Inner to Paddle Frontend

### LINKED ISSUE #22761: inner

### PR DESCRIPTION
**Pull request by SamiarAshtiany**
Close #22761

Added inner function to paddle frontend in both ivy and ivy_test

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py [hunk 1]
```diff
@@ -1003,6 +1003,39 @@ def test_paddle_heaviside(
...
 
 
+# inner
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.inner",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        num_arrays=2,
+        min_num_dims=1,
+        max_num_dims=1,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py:1013]: According to the documentation `inner` should work for N-D tensors, so we need to test all cases.
Other than that, looks good!

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py (hunk 1 above)
Comment (SamiarAshtiany) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py]: Thanks! Should be fixed now hopefully :)

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py]: The CI checks show all the backends are failing. I would suggest running the test locally to debug it.

**Review by AnnaTz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**